### ISSUE TITLE
[A BUG Maybe?] RAM requirement for BERT/PyTorch Pretraining

### ISSUE THREAD

**Issue by Hannibal046** [NVIDIA/DeepLearningExamples#1102]
Hi, I am wondering how much RAM memory does bert pretraining cost ? This is my results, it is going to break my system...
<img width="1138" alt="image" src="https://user-images.githubusercontent.com/38466901/160390565-48c6a6c2-ae81-4209-8ce1-c45c9065bc2e.png">

**Comment by Hannibal046** [comment]
I am using given docker. And I print the RAM usage after every optimization step by this command `print('Memory Usage：%.4f GB' % (psutil.Process(os.getpid()).memory_info().rss / 1024 / 1024 / 1024) )`. And I find it keeps growing

**Comment by sharathts** [comment]
Thank you for finding the bug and submitting a [PR](https://github.com/NVIDIA/DeepLearningExamples/pull/1104) to close the same. We will merge your changes.

**Comment by Hannibal046** [comment]
Although modification from `stats.device_stat('average_loss').add_(loss)` to `stats.device_stat('average_loss').add_(loss.clone().detach())` make the program look ok. But actually I don't what causes this considering `average_loss` and `loss` both are both on the GPU. And I can not reproduce this with a toy model.

<img width="984" alt="image" src="https://user-images.githubusercontent.com/38466901/160782048-9a3a8fdc-b180-4dc0-b3d6-f1d24160d30d.png">

**Comment by sharathts** [comment]
`loss` is a part of the graph. I believe the previous loss tensors/graph are retained requiring in memory when added to `average_loss` increasing memory usage as the training progresses. `.item()` should also do the job instead of `.clone().detach()`

**Comment by Hannibal046** [comment]
I totally agree with you and I find the reason why my toy example didn't work is that my toy model is too small and the memory metric I use `GB` is too big to capture the minor memory leakage. After upsizing the model and use `KB`, I observe the same memory leakage problem even if `loss` and `x` are on the same **GPU** device.
<img width="1431" alt="image" src="https://user-images.githubusercontent.com/38466901/160986775-5fe4d710-29cc-43e7-9175-e9b0d07c8160.png">

Also, the `.item()` works. Although	 it might looks nicer to use `.item()`, there maybe exists some problems about the performance.  I don't know the exact implementation of PyTorch to add a float python object to a GPU tensor. This toy example might helps:
<img width="601" alt="image" src="https://user-images.githubusercontent.com/38466901/160983477-0f7b7156-48b2-4c43-974c-98d93faa708b.png">
